bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/client/MagentoGraphqlClientImpl.java

```diff
@@ -0,0 +1,311 @@
+import org.apache.sling.models.annotations.injectorspecific.InjectionStrategy;
+import org.apache.sling.models.annotations.injectorspecific.ScriptVariable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.adobe.cq.commerce.core.components.client.DeniedHttpHeaders;
+import com.adobe.cq.commerce.core.components.client.MagentoGraphqlClient;
+import com.adobe.cq.commerce.core.components.services.ComponentsConfiguration;
+import com.adobe.cq.commerce.graphql.client.CachingStrategy;
+import com.adobe.cq.commerce.graphql.client.CachingStrategy.DataFetchingPolicy;
+import com.adobe.cq.commerce.graphql.client.GraphqlClient;
+import com.adobe.cq.commerce.graphql.client.GraphqlClientConfiguration;
+import com.adobe.cq.commerce.graphql.client.GraphqlRequest;
+import com.adobe.cq.commerce.graphql.client.GraphqlResponse;
+import com.adobe.cq.commerce.graphql.client.HttpMethod;
+import com.adobe.cq.commerce.graphql.client.RequestOptions;
+import com.adobe.cq.commerce.magento.graphql.Query;
+import com.adobe.cq.commerce.magento.graphql.gson.Error;
+import com.adobe.cq.commerce.magento.graphql.gson.QueryDeserializer;
+import com.adobe.cq.launches.api.Launch;
+import com.adobe.cq.wcm.launches.utils.LaunchUtils;
+import com.adobe.granite.ui.components.ds.ValueMapResource;
+import com.day.cq.commons.inherit.ComponentInheritanceValueMap;
+import com.day.cq.commons.inherit.HierarchyNodeInheritanceValueMap;
+import com.day.cq.commons.inherit.InheritanceValueMap;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+
+/**
+ * This is a wrapper class for {@link GraphqlClient}. The constructor adapts a {@link Resource} to
+ * the GraphqlClient class and also looks for the <code>magentoStore</code> property on the resource
+ * path in order to set the Magento <code>Store</code> HTTP header. This wrapper also sets the custom
+ * Magento Gson deserializer from {@link QueryDeserializer}.
+ */
+@Model(
+    adaptables = { SlingHttpServletRequest.class, Resource.class },
+    adapters = { MagentoGraphqlClient.class, MagentoGraphqlClientImpl.class },
+    cache = true)
+public class MagentoGraphqlClientImpl implements MagentoGraphqlClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MagentoGraphqlClient.class);
+
+    private SlingHttpServletRequest request;
+    private Resource resource;
+    @ScriptVariable(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private Page currentPage;
+
+    private GraphqlClient graphqlClient;
+    private RequestOptions requestOptions;
+    private List<Header> httpHeaders;
+
+    public MagentoGraphqlClientImpl(Resource resource) {
+        this.resource = resource;
+    }
+
+    public MagentoGraphqlClientImpl(SlingHttpServletRequest request) {
+        this.request = request;
+        this.resource = request.getResource();
+    }
+
+    /**
+     * @deprecated use sling models in all cases instead
+     */
+    @Deprecated
```

Comment:
If this PR is about deprecation, I wonder if there is any chance to get rid of this constructor? :-)

Reply:
Would be great, yes. But that would require touching a lot of unit tests. I would prefer to keep it. It is now in a private package and removing it will not cause a breaking change. 

We can create a follow up story to remove it and fix the unit tests.